docs: inline-css dev-action oriented (#89064)
vercel/next.js: docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx

## Trade-Offs

> - **Enable** if you use atomic CSS (Tailwind) and prioritize first-time visitor performance

Reviewer: ```suggestion
- **Enable** if you use atomic CSS (like Tailwind) and want to optimize first-load performance for new visitors
```

- **Skip** if returning visitors are common and you want them to benefit from cached stylesheets

### When Inline CSS Helps

Normally, the browser must download HTML, parse it, discover CSS `<link>` tags, then request stylesheets before it can render. Inlining [eliminates this request waterfall](https://web.dev/learn/performance/optimize-resource-loading#inline_critical_css), so that styles arrive with the HTML, so the browser can render immediately.

> This benefit is strongest when:

Reviewer: ```suggestion
This benefit is strongest with:
```

- **First-time visitors**: Since CSS files are render-blocking, inlining eliminates the initial download delay that first-time visitors experience. Returning visitors with cached stylesheets won't see this benefit.

- **Performance metrics**: By removing additional network requests for CSS files, inlining can significantly improve [First Contentful Paint (FCP)](https://web.dev/articles/fcp) and [Largest Contentful Paint (LCP)](https://web.dev/articles/lcp).

- **Slow connections**: For users on high-latency networks, each additional request adds delay. Inlining reduces round trips, which matters most when connections are slow.

- **Atomic CSS (Tailwind)**: Utility-first frameworks generate only the classes you use, keeping CSS small. The styles for a page don't grow proportionally with page complexity—they're typically compact regardless of how much UI you build. This makes inlining practical since you get the performance benefit without significantly bloating HTML.

### When External CSS is Better

Inlined styles cannot be cached separately from HTML. Every page load re-downloads the same CSS.

> This cost is highest when:

Reviewer: ```suggestion
This trade-off matters most with:
```
